### PR TITLE
Switch to new license Manager

### PR DESCRIPTION
**Pull request by micbar**
## Description
- Switch to new license manager
- Raise minimum dependency to 10.5

## To do

- [x] Fix tests 

@pmaier1

### CODE REVIEW

**Comment by phil-davis** [comment]
In the description: "Run acceptance tests on php7.2 an php7.4"

That is already happening - codestyle, phpunit and acceptance tests are all running on PHP 7.2, 7,3, 7.4
So that point is not happening in this PR.

**Comment by micbar** [comment]
@phil-davis I needed to remove the tests on `latest`

We need to re enable after the core release

**Comment by phil-davis** [comment]
> @phil-davis I needed to remove the tests on `latest`
> 
> We need to re enable after the core release

:+1: this had to be done on all repos that have min-version 10.5

Some time after 10.5.0 release becomes "latest" core, there will be a dozen repos to have tests against "latest" core enabled again.

**Comment by micbar** [comment]
@phil-davis Created #348

File: lib/AppConfig.php [hunk 1]
```diff
@@ -86,7 +89,7 @@ public function setUserValue($userId, $key, $value) {
 	 * @return bool
 	 */
 	public function enterpriseFeaturesEnabled() {
-		if (!$this->appManager->isEnabledForUser('enterprise_key') && !\getenv('CI')) {
+		if (!$this->licenseManager->checkLicenseFor($this->appName)) {
```
Comment (mrow4a) [inline lib/AppConfig.php:92]: `&& !\getenv('CI')` still required

Comment (micbar) [reply lib/AppConfig.php:92]: Not anymore. We have the grace period now which works also in CI

Comment (phil-davis) [reply lib/AppConfig.php:92]: For reference: https://github.com/owncloud/core/blob/master/tests/acceptance/run.sh#L774
Starts tests with the grace period popup disabled.

**Comment by mrow4a** [comment]
Weird I cannot use the app with new License Manager:
```
➜  octest git:(master) ✗ ls lib/private/License   
BasicLicense.php   ILicense.php       LicenseFetcher.php LicenseManager.php MessageService.php
```

I get this

```
Exception: {\"Exception\":\"OCP\\\\AppFramework\\\\QueryException\",\"Message\":\"Could not resolve uid! Class uid does not exist or error resolving constructor arguments: Could not resolve LicenseManager! Class LicenseManager does not exist\
```

I will try to figure out what is wrong with your PR

**Comment by mrow4a** [comment]
@micbar @pmaier1 fixed richdocuments, but requires merging https://github.com/owncloud/core/pull/37728

Tested with secure view
Zrzut ekranu 2020-07-26 o 10 26 17

**Comment by micbar** [comment]
Core PR is merged. The new daily master tarballs are build this night.

So tomorrow the CI should pass.

**Comment by phil-davis** [comment]
Rebased so CI includes the test change in #349

**Pull request merged**